docs/building-your-quest/repository.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Start Repository
+parent: Building Your Quest
+---
+
+# Start Repository
+
+Each quest in Wilco is associated with a start repository. When the user starts a new quest, his repository will be initialized using the start repository defined on the quest. The repository must be public.
+
+If start repository was not defined in `quest.yml`, The [Anythink Market] repository will be used.
+
+A few examples of reasons to use a start repository could be:
+1. Adding a bug to the code that the user will be asked to fix.
+2. Adding a new feature that the user will be asked to extend.
+3. Changing UI design to add your own branding.
+4. Using a codebase that is not related to Anythink.
+
+## Using frameworks specific files
+Users in Wilco select their frontend and backend frameworks/languages. It is common to create a start repository that allows playing a quest with several frameworks. This is also the case in the Anythink Market repository.
+
+Frameworks specific files are placed in `.framework` directory in the following structure:
```

Comment:
We need to split it to frontend and backebd

Reply:
Why? The whole idea is to not do that. I prefer to not have this restriction on the repository. Anythink have frontend and backend, but quests might have only one of them or even none of them. It will allow much more freedom and simplify the Engine code

Reply:
How do I know which framework is backend and which frontend?

Reply:
As long as we won't have same frameworks in both backend and frontend, we don't really care. And as we have control over the frameworks, we could make sure this won't happen. Eventually we just have two possible folder names and we use them to figure out which files to use. At this point, he flexibility of not assuming backend and frontend totally worth it imho. It is definitely makes sense for companies in which I am pretty certain they would not want this specific structure.

Reply:
So we'll have a map for each framework if it's a frontend or a backend? And we're going to put there all the available frameworks?

Reply:
Why a map? We really should not care in the repository structure what is frontend and what is backend. It is eventually just a list of frameworks, each having its relevant directories/files. I want to decouple the repository structure from the way we currently allow users to set frameworks and from specific project structures. Note that there is no must to user the frameworks folder and files can just be added to the root folder

Reply:
In anythink market we gonna have react, python, node and rails
How's the generated selection will be look like?

Reply:
This is the repo structure: https://github.com/trywilco/Anythink-Market-Public/tree/feat/new_structure/.framework.

If user's selected frameworks are react and rails, for example, we will use the files under 'react' (currently just the frontend directory) and the files under 'rails' (backend directory, docker-compose and charts).

Reply:
And we they have Go?

Reply:
We will Just add GO folder with the relevant backend, charts and docker-compose (BTW, as some point we need to change docker-compose to be non framework specific so that framework specific stuff will move to the Dockerfile files)

Reply:
But how they'll be able to choose Go?
Does the list of framework is auto generated or static?
Does users choose it once or per quest?

Reply:
We already have frameworks specification in `quest.yml`. Currently I think it has to be one of our specific frameworks, but we should be easily able to let users select the frameworks according to what quest allows. Your questions are totally valid but are also very much product questions. I think the best thing for us will be to use the frameworks defined on the `quest.yml` and users could select the frameworks when starting a quest and could also search for quests using specific frameworks.

Reply:
Ah, ok, the framework list in the quest is definitely close to loop
I thought maybe we can drop that but we can use this for now